File: allinone-kotlin/app/src/main/java/com/example/allinone/data/Note.kt
package com.example.allinone.data

import java.util.Date

data class Note(
    val id: Long = 0,
    val title: String,
    val content: String,
    val date: Date = Date(),
    val imageUris: String? = null,  // Comma-separated list of image URIs
    val videoUris: String? = null,  // Comma-separated list of video URIs
    val voiceNoteUris: String? = null,  // Comma-separated list of voice note URIs
    val lastEdited: Date = Date(),
    val isRichText: Boolean = true  // Flag to indicate if content has rich formatting
) 

File: src/types/Note.ts
export interface Note {
  id: number;
  title: string;
  content: string;
  date: string; // ISO string format for serialization
  imageUris?: string; // Comma-separated list of image URIs
  videoUris?: string; // Comma-separated list of video URIs
  voiceNoteUris?: string; // Comma-separated list of voice note URIs
  drawingUris?: string; // Comma-separated list of drawing URIs
  lastEdited: string; // ISO string format for serialization
  isRichText: boolean; // Flag to indicate if content has rich formatting
}

export interface NoteFormData {
  title: string;
  content: string;
  imageUris?: string;
  videoUris?: string;
  voiceNoteUris?: string;
  drawingUris?: string;
}

 